packages/playground/website/src/components/ensure-playground-site/ensure-playground-site-is-selected.tsx

```diff
@@ -82,13 +84,22 @@ export function EnsurePlaygroundSiteIsSelected({
 				return;
 			}
 
+			// If only the 'modal' parameter changes in searchParams, don't reload the page
+			const notRefreshingParam = 'modal';
+			const oldParams = new URLSearchParams(prevUrl?.search);
+			const newParams = new URLSearchParams(url?.search);
+
+			if ((!oldParams.has(notRefreshingParam) && newParams.has(notRefreshingParam)) || (oldParams.has(notRefreshingParam) && !newParams.has(notRefreshingParam))) {
+				return;
+			}
```

Comment:
This confirms that the `modal` parameter was added or removed from the params, but it doesn't prove that `modal` is the only param that changed.

What if we say, "Don't reload the page if nothing changed other than the `modal` param"? Because URLSearchParams should preserve parameter order and render the query strings consistently, I think we can say that like this:

```suggestion
			const oldParams = new URLSearchParams(prevUrl?.search);
			const newParams = new URLSearchParams(url?.search);
			oldParams.delete('modal');
			newParams.delete('modal');

			if (oldParams.toString() === newParams.toString()) {
				return;
			}
```

That way, we ignore changes to the `modal` param while reloading for all other param changes.

Reply:
> What if we say, "Don't reload the page if nothing changed other than the `modal` param"? 

😄 sorry, that always the intent here. It didn't need restated but was intended to be in contrast to the old logic, which said "Don't reload the page if `modal` was in one set but not the other."

Reply:
Thanks for pointing this out @brandonpayton ! 🙏🏻 

However, the more I think about this change, the more I'm afraid that it might introduce unintended bugs.

For example, I imagine a situation where we want to display only the information modal, but the remaining parameters are to cause reloading and loading a new playground (e.g. with a new blueprint json, or maybe parameters for PR preview?)

What do you think about this?
I'm asking because I don't know, both solutions have their advantages 😅

Reply:
If you received the previous comment via email, please ignore it. 
I had a coffee and realized that this is a good fix. 😅 

Changes ready for review.

Reply:
😄 all good!